### PR TITLE
Dev 3

### PR DESCRIPTION
**Pull request by IvanChepik**
Program for optimize productivity and coast

### CODE REVIEW

File: DEV-3/DEV-3/Company.cs [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        /// in final project list.
+        /// </summary>
+        private void CountQuantityEmp()
+        {
+            foreach (var emp in Optimization.EmployeesToWork)
+            {               
+                if (emp is Lead)
+                {
+                    LeadsOnProject++;
+                    continue;
```
Comment (AlexeiRomankov) [inline DEV-3/DEV-3/Company.cs]: do not use continue

File: DEV-3/DEV-3/Company.cs [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                }
+
+                if (emp is Senior)
```
Comment (AlexeiRomankov) [inline DEV-3/DEV-3/Company.cs]: is it possible use else if or switch case

File: DEV-3/Models/Employee.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿namespace Models
+{
+    /// <summary>
+    /// Abstract class Employee
+    /// this class included common fields of all employes.
+    /// </summary>
+    public abstract class Employee
+    {
+        public abstract decimal Sum { get; }
```
Comment (AlexeiRomankov) [inline DEV-3/Models/Employee.cs]: do you really need abstract properties?

Comment (AlexeiRomankov) [inline DEV-3/Models/Employee.cs]: Unclear variable Sum. Is it Salary?

File: DEV-3/Models/Employee.cs [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+        public  abstract int Productivity { get;  }
+        public decimal Coef => Productivity / Sum;
```
Comment (AlexeiRomankov) [inline DEV-3/Models/Employee.cs]: Unclear variable Coef

File: DEV-3/DEV-3/EntryPoint.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        }
+
+        /// <summary>
+        /// Method TryParseCriterion
+        /// select type of object which will in ref criterion by argument choosingCriterion.
+        /// </summary>
+        /// <param name="choosingCriterion">number which select type of criterion </param>
+        /// <param name="criterion">ref on criterion which was selected in this method</param>
+        /// <returns>return true if parsing is correct or false if is not correct</returns>
+        private static bool TryParseCriterion(int choosingCriterion, out IOptimize criterion)
```
[the remark this answers is not in the document]
Comment (IvanChepik) [reply DEV-3/DEV-3/EntryPoint.cs]: It's a standart template for TryXXX method. I need to choose criterion

**Pull request merged**